benchmarks/naslib/problem.py

```diff
@@ -0,0 +1,143 @@
+        )
+
+    def create_problem(self, seed: int) -> problem.Problem:
+        return NASLibProblem(self._search_space, self._converter, **self._config)
+
+
+class NASLibProblem(problem.Problem):
+    def __init__(self, search_space: Graph, converter: Any, direction: str, **config: Any) -> None:
+        """"""
+        # params
+        # metric      : Metric to query for
+        # dataset     : Dataset to query for
+        # epoch       : If specified, returns the metric of the arch at that epoch of training
+        # full_lc     : If true, returns the curve of the given metric in all epochs
+        # dataset_api : API to use for querying metrics
+        # hp          : Number of epochs the model was trained for. Value is in {1, 12, 90}
+        # is_random   : When True, the performance of a random architecture will be returned
+        #               When False, the performanceo of all trials will be averaged.
+        super().__init__()
+        self._search_space = search_space
+        self._config = config
+        self._converter = converter
+        self._scale = 1 if direction == "minimize" else -1
+
+    def create_evaluator(self, params: List[Any]) -> problem.Evaluator:
+        return NASLibEvaluator(
+            self._converter(params), self._search_space.copy(), self._scale, self._config
+        )
+
+
+class NASLibEvaluator(problem.Evaluator):
+    def __init__(
+        self,
+        params: List[Optional[float]],
+        search_space: Graph,
+        scale: float,
+        config: Dict[str, Any],
+    ) -> None:
+        self._search_space = search_space
+        # print(params, file=sys.stderr)
+        self._search_space.set_spec(params)
+        self._config = config
+        self._current_step = 0
+        self._scale = scale
+        self._lc = self._search_space.query(**self._config)
+
+    def current_step(self) -> int:
+        return self._current_step
+
+    def evaluate(self, next_step: int) -> List[float]:
+        self._current_step = next_step
+        return [self._lc[next_step] * self._scale]
+
+
+if __name__ == "__main__":
+    sys.stdout.close()
+    sys.stdout = sys.__stdout__
+
+    optimize_direction = {
+        "TRAIN_ACCURACY": "maximize",
+        "VAL_ACCURACY": "maximize",
+        "TEST_ACCURACY": "maximize",
+        "TRAIN_LOSS": "minimize",
+        "VAL_LOSS": "minimize",
+        "TEST_LOSS": "minimize",
+        "TRAIN_TIME": "minimize",
+        "VAL_TIME": "minimize",
+        "TEST_TIME": "minimize",
+        "FLOPS": "minimize",
+        "LATENCY": "minimize",
+    }
+
+    default_config = {"metric": "TEST_ACCURACY", "epoch": -1, "full_lc": True}
+
+    if len(sys.argv) < 1 + 2:
+        print("Usage: python3 nas_bench_suite/problems.py <search_space> <dataset> [<config>]")
+        print(
+            "Example: python3 nas_bench_suite/problems.py nasbench201"
+            " cifar10 \"{'metric':'TEST_ACCURACY'}\""
+        )
+        exit(1)
```

Comment:
It may be overengineering to take the `config` argument, since we need to expect deep understanding of NASLib. How about removing it in this PR and working on it in a follow-up PR.